### PR TITLE
MNTOR-2216: FxA webhook deletion should remove OneRep data

### PR DESCRIPTION
**Pull request by mansaj**
# References: 
Jira: MNTOR-2216

# Description
* Modify FxA webhook deletion to include OneRep removal
* Cascade delete for one rep tables etc.

# Screenshot (if applicable)

Not applicable.

# How to test
In heroku or stage, try deleting the user in fxa and see it the event logged and triggering a cascade delete. Verify that existing OneRep records don't exist anymore for that subscriber 

# Checklist (Definition of Done)
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] I've added or updated the relevant sections in readme and/or code comments
- [x] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [x] All acceptance criteria are met.
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.
- [x] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: src/app/api/v1/fxa-rp-events/route.ts [hunk 1]
```diff
@@ -177,15 +177,38 @@ export async function POST(request: NextRequest) {
...
-      case FXA_DELETE_USER_EVENT:
+      case FXA_DELETE_USER_EVENT: {
         logger.info("fxa_delete_user", {
           subscriber: subscriber.id,
           event,
         });
 
+        // get profile id
+        const result = await getOnerepProfileId(subscriber.id);
+        const oneRepProfileId = result?.[0]?.["onerep_profile_id"] as number;
```
Comment (rhelmer) [inline src/app/api/v1/fxa-rp-events/route.ts:188]: Would you mind filing a bug to refactor this? It'd be nice if the caller just got a `number` back.

File: src/app/api/v1/fxa-rp-events/route.ts [hunk 2]
```diff
@@ -177,15 +177,38 @@ export async function POST(request: NextRequest) {
...
+              });
+          }
+
+          logger.info("deactivated_onerep_profile", {
+            subscriber_id: subscriber.id,
+          });
+        }
+
         // delete user events only have keys. Keys point to empty objects
         await deleteSubscriber(subscriber);
```
Comment (rhelmer) [inline src/app/api/v1/fxa-rp-events/route.ts:209]: OK so this does the deletion and just cascades now right? I think this makes sense, the migration is a bit tough to read but if that's the intention it lgtm.

**Review by rhelmer — APPROVED** [review]

File: src/app/api/v1/fxa-rp-events/route.ts (hunk 2 above)
Comment (mansaj) [reply src/app/api/v1/fxa-rp-events/route.ts:209]: yes

File: src/app/api/v1/fxa-rp-events/route.ts (hunk 1 above)
Comment (mansaj) [reply src/app/api/v1/fxa-rp-events/route.ts:188]: I think I can just refactor that in this PR

Comment (mansaj) [reply src/app/api/v1/fxa-rp-events/route.ts:188]: actually let me open another PR for it..

Comment (mansaj) [reply src/app/api/v1/fxa-rp-events/route.ts:188]: https://mozilla-hub.atlassian.net/browse/MNTOR-2874

**Pull request merged**